spring-cloud-contract-tools/spring-cloud-contract-spec-pact/src/main/groovy/org/springframework/cloud/contract/verifier/spec/pact/MessagingSCContractCreator.groovy

```diff
@@ -0,0 +1,136 @@
+package org.springframework.cloud.contract.verifier.spec.pact
+
+import au.com.dius.pact.model.OptionalBody
+import au.com.dius.pact.model.matchingrules.Category
+import au.com.dius.pact.model.matchingrules.DateMatcher
+import au.com.dius.pact.model.matchingrules.MatchingRule
+import au.com.dius.pact.model.matchingrules.MatchingRuleGroup
+import au.com.dius.pact.model.matchingrules.MaxTypeMatcher
+import au.com.dius.pact.model.matchingrules.MinMaxTypeMatcher
+import au.com.dius.pact.model.matchingrules.MinTypeMatcher
+import au.com.dius.pact.model.matchingrules.RegexMatcher
+import au.com.dius.pact.model.matchingrules.TimeMatcher
+import au.com.dius.pact.model.matchingrules.TimestampMatcher
+import au.com.dius.pact.model.matchingrules.TypeMatcher
+import au.com.dius.pact.model.v3.messaging.Message
+import au.com.dius.pact.model.v3.messaging.MessagePact
+import groovy.json.JsonException
+import groovy.json.JsonSlurper
+import groovy.transform.CompileStatic
+import groovy.transform.PackageScope
+import org.springframework.cloud.contract.spec.Contract
+import org.springframework.cloud.contract.verifier.util.JsonPaths
+import org.springframework.cloud.contract.verifier.util.JsonToJsonPathsConverter
+
+/**
+ * Creator of {@link Contract} instances
+ *
+ * @author Tim Ysewyn
+ * @since 2.0.0
+ */
+@CompileStatic
+@PackageScope
+class MessagingSCContractCreator {
+
+	private static final JsonSlurper jsonSlurper = new JsonSlurper()
+	private static final String FULL_BODY = '$'
+
+	Collection<Contract> convertFrom(MessagePact pact) {
+		pact.messages.collect({ Message message ->
+			Contract.make {
+				label("$message.description")
+
+				if (!message.providerStates.isEmpty()) {
+					input {
+						triggeredBy(getTriggeredBy(message))
+					}
+				}
+
+				outputMessage {
+					if (message.contents.present) {
+						body(parseBody(message.contents))
+
+
+						Category bodyRules = message.matchingRules.rulesForCategory('body')
+						if (bodyRules && !bodyRules.matchingRules.isEmpty()) {
```

Comment:
matching rules can't be null ? I smell NPE ;)

Reply:
No, the matchingRules parameter will be set by default on the `Message` object.
But see this [comment](https://github.com/spring-cloud/spring-cloud-contract/pull/569/files/bb44f55cf028564e82f0ab6222a67d0627d56087#r175776041)